src/Roslyn.Diagnostics.Analyzers/Core/RelaxTestNamingSuppressor.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Concurrent;
+using System.Collections.Immutable;
+using Analyzer.Utilities;
+using Analyzer.Utilities.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Roslyn.Diagnostics.Analyzers.Extensions;
+
+namespace Roslyn.Diagnostics.Analyzers
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    public sealed class RelaxTestNamingSuppressor : DiagnosticSuppressor
+    {
+        private const string Id = RoslynDiagnosticIds.RelaxTestNamingSuppressionRuleId;
+        private const string SuppressedDiagnosticId = "VSTHRD200";
+        private static readonly LocalizableString s_localizableJustification = new LocalizableResourceString(nameof(RoslynDiagnosticsAnalyzersResources.RelaxTestNamingSuppressorJustification), RoslynDiagnosticsAnalyzersResources.ResourceManager, typeof(RoslynDiagnosticsAnalyzersResources));
+
+        internal static readonly SuppressionDescriptor Rule =
+            new SuppressionDescriptor(Id, SuppressedDiagnosticId, s_localizableJustification);
+
+        public override ImmutableArray<SuppressionDescriptor> SupportedSuppressions { get; } = ImmutableArray.Create(Rule);
+
+        public override void ReportSuppressions(SuppressionAnalysisContext context)
+        {
+            if (!(context.Compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.XunitFactAttribute) is { } factAttribute))
```

Comment:
I am probably not using the right keywords but I can't find the explanation of the `XXX is {} name` syntax, could you enlighten me?

Reply:
It's like `var name`, except `{ } name` is only true when `name` is non-null.